src/client.ts

```diff
@@ -207,6 +212,10 @@ export class APIClient {
 
     const malwares = body.data.malwares;
 
+    // TODO: Only keep the latest if there are duplicates
+    // If there is a duplicate of a malware and they share the same guid and machineName
+    // Only keep the one that has the highest timestamp and don't iterate over the others
+
```

Comment:
This makes sense but I am worried that this would require lots of iteration over the data that we receive. I can see this affecting the _'speed'_ drastically. Is that a concern for us?

Reply:
I'm afraid this is necessary, the reason being is the graph data is supposed to represent the current snapshot of all the resources/entities, it shouldn't hold some historical data in it, but this is not a problem that we're often faced with. Whatever we decide to do we'll check back with JupiterOne later and see if it requires further changes.

Options that we have currently are:
1) Make timestamp be part of the key. This would result in no duplicates, but if we ever later need to find some malware via its key, it might be impossible (e.g. we might know the guid of malware but not the timestamp)
2) The filtering idea I added in the comments.

In my opinion, just go with that filtering option for and we'll see if it'll require some changes later down the road.